src/core/lib/event_engine/posix_engine/traced_buffer_list.cc

```diff
@@ -0,0 +1,289 @@
+        metrics->delivery_rate = ReadUnaligned<uint64_t>(val);
+        break;
+      }
+      case TCP_NLA_DELIVERY_RATE_APP_LMT: {
+        metrics->is_delivery_rate_app_limited = ReadUnaligned<uint8_t>(val);
+        break;
+      }
+      case TCP_NLA_SND_CWND: {
+        metrics->congestion_window = ReadUnaligned<uint32_t>(val);
+        break;
+      }
+      case TCP_NLA_MIN_RTT: {
+        metrics->min_rtt = ReadUnaligned<uint32_t>(val);
+        break;
+      }
+      case TCP_NLA_SRTT: {
+        metrics->srtt = ReadUnaligned<uint32_t>(val);
+        break;
+      }
+      case TCP_NLA_RECUR_RETRANS: {
+        metrics->recurring_retrans = ReadUnaligned<uint8_t>(val);
+        break;
+      }
+      case TCP_NLA_BYTES_SENT: {
+        metrics->data_sent = ReadUnaligned<uint64_t>(val);
+        break;
+      }
+      case TCP_NLA_DATA_SEGS_OUT: {
+        metrics->packet_sent = ReadUnaligned<uint64_t>(val);
+        break;
+      }
+      case TCP_NLA_TOTAL_RETRANS: {
+        metrics->packet_retx = ReadUnaligned<uint64_t>(val);
+        break;
+      }
+      case TCP_NLA_DELIVERED: {
+        metrics->packet_delivered = ReadUnaligned<uint32_t>(val);
+        break;
+      }
+      case TCP_NLA_DELIVERED_CE: {
+        metrics->packet_delivered_ce = ReadUnaligned<uint32_t>(val);
+        break;
+      }
+      case TCP_NLA_BYTES_RETRANS: {
+        metrics->data_retx = ReadUnaligned<uint64_t>(val);
+        break;
+      }
+      case TCP_NLA_DSACK_DUPS: {
+        metrics->packet_spurious_retx = ReadUnaligned<uint32_t>(val);
+        break;
+      }
+      case TCP_NLA_REORDERING: {
+        metrics->reordering = ReadUnaligned<uint32_t>(val);
+        break;
+      }
+      case TCP_NLA_SND_SSTHRESH: {
+        metrics->snd_ssthresh = ReadUnaligned<uint32_t>(val);
+        break;
+      }
+    }
+    offset += NLA_ALIGN(attr->nla_len);
+  }
+}
+}  // namespace.
+
+void TracedBuffer::AddNewEntry(std::list<TracedBuffer*>& traced_buffers,
+                               uint32_t seq_no, int fd, void* arg) {
+  TracedBuffer* new_elem = new TracedBuffer(seq_no, arg);
+  // Store the current time as the sendmsg time.
+  new_elem->ts_.sendmsg_time.time = gpr_now(GPR_CLOCK_REALTIME);
+  new_elem->ts_.scheduled_time.time = gpr_inf_past(GPR_CLOCK_REALTIME);
+  new_elem->ts_.sent_time.time = gpr_inf_past(GPR_CLOCK_REALTIME);
+  new_elem->ts_.acked_time.time = gpr_inf_past(GPR_CLOCK_REALTIME);
+
+  if (GetSocketTcpInfo(&new_elem->ts_.info, fd) == 0) {
+    ExtractOptStatsFromTcpInfo(&new_elem->ts_.sendmsg_time.metrics,
+                               &new_elem->ts_.info);
+  }
+
+  traced_buffers.push_back(new_elem);
+}
+
+void TracedBuffer::ProcessTimestamp(std::list<TracedBuffer*>& traced_buffers,
+                                    struct sock_extended_err* serr,
+                                    struct cmsghdr* opt_stats,
+                                    struct scm_timestamping* tss) {
+  auto it = traced_buffers.begin();
```

Comment:
This can be `auto& it`

Reply:
This is giving a "Non-const lvalue reference to type '_List_iterator<...>' cannot bind to a temporary of type" compile time error